src/diffusers/pipelines/audioldm2/pipeline_audioldm2_tts.py

```diff
@@ -0,0 +1,1002 @@
+                be computed from `prompt` input argument.
+            negative_attention_mask (`torch.LongTensor`, *optional*):
+                Pre-computed attention mask to be applied to the `negative_prompt_embeds`. If not provided, attention
+                mask will be computed from `negative_prompt` input argument.
+            max_new_tokens (`int`, *optional*, defaults to None):
+                The number of new tokens to generate with the GPT2 language model.
+        Returns:
+            prompt_embeds (`torch.FloatTensor`):
+                Text embeddings from the Flan T5 model.
+            attention_mask (`torch.LongTensor`):
+                Attention mask to be applied to the `prompt_embeds`.
+            generated_prompt_embeds (`torch.FloatTensor`):
+                Text embeddings generated from the GPT2 langauge model.
+
+        ```"""
+        if prompt is not None and isinstance(prompt, str):
+            batch_size = 1
+        elif prompt is not None and isinstance(prompt, list):
+            batch_size = len(prompt)
+        else:
+            batch_size = prompt_embeds.shape[0]
+
+        # Define tokenizers and text encoders
+        tokenizers = [self.tokenizer, self.tokenizer_2]
+
+        text_encoders = [self.text_encoder, self.text_encoder_2.text_encoder]
+
+        if prompt_embeds is None:
+            prompt_embeds_list = []
+            attention_mask_list = []
+
+            for tokenizer, text_encoder in zip(tokenizers, text_encoders):
+                if text_encoder.config.model_type == "clap":
+                    text_inputs = tokenizer(
+                        prompt,
+                        padding="max_length"
+                        if isinstance(tokenizer, (RobertaTokenizer, RobertaTokenizerFast))
+                        else True,
+                        max_length=tokenizer.model_max_length,
+                        truncation=True,
+                        return_tensors="pt",
+                    )
+
+                    text_input_ids = text_inputs.input_ids
+                    attention_mask = text_inputs.attention_mask
+                    untruncated_ids = tokenizer(prompt, padding="longest", return_tensors="pt").input_ids
+
+                    if untruncated_ids.shape[-1] >= text_input_ids.shape[-1] and not torch.equal(
+                        text_input_ids, untruncated_ids
+                    ):
+                        removed_text = tokenizer.batch_decode(untruncated_ids[:, tokenizer.model_max_length - 1 : -1])
+                        logger.warning(
+                            f"The following part of your input was truncated because {text_encoder.config.model_type} can "
+                            f"only handle sequences up to {tokenizer.model_max_length} tokens: {removed_text}"
+                        )
+
+                else:
+                    # Convert text to phoneme
+                    phonemes = self.text_to_phoneme(transcription)
```

Comment:
The VITS tokenizer should handle the phoneme conversion no? _c.f._ https://github.com/huggingface/transformers/blob/14b04b4b9c483d94fadd2b5479ed9430bae8ac84/src/transformers/models/vits/tokenization_vits.py#L199

Looks like the tokenizer has `phonemize` set to False: https://huggingface.co/anhnct/audioldm2_giga_speech/blob/2b8d87e455975a706cda5d9354100c611899bb91/tokenizer_2/tokenizer_config.json#L20

You can pass `phonemeize=True` when you instantiate the tokenizer in the conversion script and this will enable the tokenizer's phonemization